contrib/lkt/language/parser.py

```diff
@@ -574,6 +574,8 @@ class Decl(LKNode):
         """,
     )
 
+    decl_type_name = Property((S("<decl_type>")))
```

Comment:
> Also, shouldn’t it be abstract, rather than returning a dummy value here?

This is a matter of taste I guess. Making it abstract requires to define it for all remaining Decl subclasses: GrammarDecl, GrammarRuleDecl, SyntheticLexerDecl, LexerDecl, LexerFamilyDecl, EnumClassAltDecl, FunctionType, GenericFormalTypeDecl, InstantiatedGenericType.

What do you prefer?

Reply:
I don’t think it’s a matter of taste: if the property is not abstract, we’ll get bogous error messages if we forget to override the property in several classes (GrammarDecl and the others, which is precisely the case here). If the property is abstract, Langkit will force us to do the overriding everywhere needed.

I’m definitely in favor of using the typing system to help us write correct/complete code. ;-)

Reply:
No problem, it is now abstract!